### PR TITLE
DataViews table layout: hide actions menu when there is only one action and is primary

### PR DESCRIPTION
**Pull request by oandregal**
Part of https://github.com/WordPress/gutenberg/issues/65165

## What?

This PR hides the actions menu for the DataViews table layout when there's only 1 action and is primary.

| Before | After |
| --- | --- |
| Screenshot 2024-11-15 at 08 00 30 | Screenshot 2024-11-15 at 08 00 54 |

_All current core screens have more than one action. See testing instructions below._ 

## Why?

To improve UX.

## Testing Instructions

This is something we've heard from extenders. Because there's no place in core where we have this situation just yet, we have to create one for testing:

- Go to the [post-list code](https://github.com/WordPress/gutenberg/blob/hide/actions-menu-list/packages/edit-site/src/components/post-list/index.js#L350) and remove the `postTypeActions` in this line and in the line below (in both arrays).
- Visit the Pages page and switch to the table layout.
- Verify the actions menu is hidden, only the edit icon is displayed.

### CODE REVIEW

**Review by Mamaduka — APPROVED** [review]
Works as advertised ✅

File: packages/dataviews/src/components/dataviews-item-actions/index.tsx [hunk 1]
```diff
@@ -179,6 +185,13 @@ export function ActionsMenuGroup< Item >( {
 	);
 }
 
+function hasOnlyOneActionAndIsPrimary< Item >(
+	primaryActions: Action< Item >[],
+	actions: Action< Item >[]
+) {
+	return primaryActions.length === 1 && actions.length;
```
Comment (ntsekouras) [inline packages/dataviews/src/components/dataviews-item-actions/index.tsx:192]: Why not inline the check?

**Comment by ntsekouras** [comment]
Noting that we do this for every layout and not only for `table`. We should update the title.

I was a bit confused that `list` layout doesn't use these action components. It seems probably about adding some extra things for keyboard navigation and I'm wondering if we can absorb that somehow.

**Comment by oandregal** [comment]
> Noting that we do this for every layout and not only for table. We should update the title.

This is actually only for the table. Here's the PR for the list https://github.com/WordPress/gutenberg/pull/67015 :)

The list view uses the subcomponents directly to handle the keyboard interactions. The grid is unaffected because it uses the compact view (no primary actions visible). This can be tested by switching to any other layout during testing.

| List | Grid | Table |
| --- | --- | --- |
| Screenshot 2024-11-15 at 09 28 18 |  Screenshot 2024-11-15 at 09 28 09 | Screenshot 2024-11-15 at 09 28 02 |

File: packages/dataviews/src/components/dataviews-item-actions/index.tsx (hunk 1 above)
Comment (oandregal) [reply packages/dataviews/src/components/dataviews-item-actions/index.tsx:192]: That could work as well. No big feelings either way. I suppose I thought it better to have it explain the  _why_.

**Comment by ntsekouras** [comment]
>The list view uses the subcomponents directly to handle the keyboard interactions. The grid is unaffected because it uses the compact view (no primary actions visible). This can be tested by switching to any other layout during testing.

Exactly. If grid wasn't using the compact and had one primary action would have the same behavior. For now it doesn't, but when we open the API for registering layouts, this behavior will be for any layout.

**Comment by oandregal** [comment]
@ntsekouras Agreed. If things worked differently then yes, it'd make sense to change the changelog/issue title to be about all the layouts. I just worry that it'd be confusing to say so when it's not true right now? Not sure how it could be improved without being confusing, but happy to explore alternatives if you have any suggestion.

**Comment by ntsekouras** [comment]
The title is not that important right now, yes. We should keep this in mind when creating the API for layouts to document it.

File: packages/dataviews/src/components/dataviews-item-actions/index.tsx (hunk 1 above)
Comment (okmttdhr) [inline packages/dataviews/src/components/dataviews-item-actions/index.tsx:192]: @oandregal I believe the condition `actions.length` should be updated to `actions.length === 1`. The current implementation only displays the primary action when there is exactly one primary action, even if additional non-primary actions are present.

https://wordpress.github.io/gutenberg/?path=/story/dataviews-dataviews--default
Screenshot 2024-11-21 at 17 01 42
https://github.com/WordPress/gutenberg/blob/b7d989e0dafb23866f65295a851e6b8734c3b693/packages/dataviews/src/components/dataviews/stories/fixtures.tsx#L610-L615

Comment (oandregal) [reply packages/dataviews/src/components/dataviews-item-actions/index.tsx:192]: Thanks, @okmttdhr! Prepared a fix at https://github.com/WordPress/gutenberg/pull/67197 The fix will be part of 19.8, as this PR is, which hasn't been released yet.

**Pull request merged**